Makefile

```diff
@@ -821,6 +813,56 @@ ifneq ($(word 2,$(MAKECMDGOALS)),)
 export ENABLE_MSG_EXTRA := yes
 endif
 
+uavobjects_armsoftfp: TARGET=uavobjects_armsoftfp
+uavobjects_armsoftfp: OUTDIR=$(BUILD_DIR)/$(TARGET)
+uavobjects_armsoftfp: uavobjects
+	$(V1) mkdir -p $(OUTDIR)/dep
+	$(V1) cd $(ROOT_DIR)/flight/uavobjectlib && \
+		$(MAKE) -r --no-print-directory \
+		TCHAIN_PREFIX="$(ARM_SDK_PREFIX)" \
+		REMOVE_CMD="$(RM)" \
+		\
+		MAKE_INC_DIR=$(MAKE_INC_DIR) \
+		ROOT_DIR=$(ROOT_DIR) \
+		TARGET=$(TARGET) \
+		OUTDIR=$(OUTDIR) \
+		\
+		PIOS=$(PIOS) \
```

Comment:
This is really only boilerplate if it gets blindly copy/pasted without being appropriately limited.  The set of paths passed into the submakefiles was originally intended to form a basic API to describe where the submakefiles should be looking/writing.  Some attempt was made to pass in only those paths that made sense for each build type.

Deleting these explicit parameters and exposing the root paths as global variables (via export) would throw away any scoping for submakefiles.  It sounds like that's your ultimate intent with this, right?  I'm not fond of that idea.

Could you seek out another reviewer with makefile experience to take a look at this particular aspect of the changes (and the related export of BUILD_DIR) and write their comments here?  I'll defer to whatever you and your 2nd reviewer decide so as to not hold up this merge any further.